test/e2e/util.go

```diff
@@ -714,6 +715,20 @@ func waitForReplicationController(c *client.Client, namespace, name string, exis
 	return nil
 }
 
+func waitForEndpoint(c *client.Client, ns, name string) error {
+	for t := time.Now(); time.Since(t) < endpointRegisterTimeout; time.Sleep(poll) {
+		endpoint, err := c.Endpoints(ns).Get(name)
+		Expect(err).NotTo(HaveOccurred())
+		if len(endpoint.Subsets) == 0 || len(endpoint.Subsets[0].Addresses) == 0 {
+			Logf("Endpoint %s/%s is not ready yet", ns, name)
+			continue
+		} else {
+			return nil
+		}
+	}
+	return fmt.Errorf("Failed to get entpoints for %s/%s", ns, name)
```

Comment:
you never check whether an error occurred or not. so if the endpoint fails to startup, the error is lost and you continue anyway.

(also nit: misspelled "endpoint")